packages/datadog-plugin-elasticsearch/src/index.js

```diff
@@ -20,48 +22,39 @@ function createWrapRequest (tracer, config) {
           'span.type': 'elasticsearch',
           'elasticsearch.url': params.path,
           'elasticsearch.method': params.method,
+          'elasticsearch.body': body,
           'elasticsearch.params': JSON.stringify(params.querystring || params.query)
         }
       })
 
-      if (params.body || params.bulkBody) {
-        span.setTag('elasticsearch.body', JSON.stringify(params.body || params.bulkBody))
-      }
-
       analyticsSampler.sample(span, config.measured)
 
-      cb = request.length === 2 || typeof options === 'function'
-        ? tracer.scope().bind(options, childOf)
-        : tracer.scope().bind(cb, childOf)
-
-      return tracer.scope().activate(span, () => {
-        if (typeof cb === 'function') {
-          if (request.length === 2) {
-            return request.call(this, params, wrapCallback(tracer, span, params, config, cb))
-          } else {
-            return request.call(this, params, options, wrapCallback(tracer, span, params, config, cb))
-          }
-        } else {
-          const promise = request.apply(this, arguments)
+      cb = arguments[lastIndex]
 
-          if (promise && typeof promise.then === 'function') {
-            promise.then(() => finish(span, params, config), e => finish(span, params, config, e))
-          } else {
-            finish(span, params, config)
-          }
+      if (typeof cb === 'function') {
+        arguments[lastIndex] = wrapCallback(tracer, span, params, config, cb)
 
-          return promise
+        return tracer.scope().activate(span, () => request.apply(this, arguments))
+      } else {
+        const promise = request.apply(this, arguments)
+
+        if (promise && typeof promise.then === 'function') {
+          promise.then(() => finish(span, params, config), e => finish(span, params, config, e))
+        } else {
+          finish(span, params, config)
         }
-      })
+
+        return promise
+      }
     }
   }
 }
 
 function wrapCallback (tracer, span, params, config, done) {
-  return function (err) {
+  return tracer.scope().bind(function (err) {
     finish(span, params, config, err)
     done.apply(null, arguments)
-  }
+  })
```

Comment:
This doesn't need the `childOf` argument anymore?

Reply:
It only needs it to explicitly set the parent, otherwise it automatically uses the current scope.